### PR TITLE
Api client / composables reference docs

### LINKED ISSUE #5428: Doc: Commercetools > Composables API Reference / API Client Reference
I think that this task has three main subtasks. 

- creating a script for auto-generating API Reference,
- how exactly these sections should look like (placing, UI?),
- adding References to proper places in our docs

At the moment I have used `typedoc` (I had some problems with api-extractor) for auto-generating API Reference and I have a lot of markdown files (screenshot below).

Zrzut ekranu 2021-01-20 o 15 39 48

The next step should be adding all(?) these files to proper places.

### PR DESCRIPTION
**Pull request by lukaszjedrasik**
### Related Issues

closes #5428 

### Short Description and Why It's Useful
As we talked, autogenerated are only API Client References. 

### Which Environment This Relates To
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

* [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
* [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
* [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
* [x] No upgrade steps required (100% backward compatibility and no breaking changes)
* [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature
* [ ] (Next only) I've followed this [instruction](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and I've created a `.js` file with information about my Pull Request

**IMPORTANT NOTICE**

* Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

* [ ] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by NataliaTepluhina** [comment]
I wonder if it would be possible to use a generated file as a source for VuePress sidebar navigation, so we could have both table of contents and a sidebar (like [here](https://v3.vuejs.org/api/))

File: packages/core/docs/commercetools/api-client-reference/modules.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+## Table of contents
+
+### Modules
+
+- [addToCart](modules/addtocart.md)
+- [applyCartCoupon](modules/applycartcoupon.md)
+- [createCart](modules/createcart.md)
+- [createCart/defaultMutation](modules/createcart_defaultmutation.md)
+- [createMyOrderFromCart](modules/createmyorderfromcart.md)
```
Comment (filrak) [inline packages/core/docs/commercetools/api-client-reference/modules.md:11]: where this `modules/createmyorderfromcart.md` is?

File: packages/core/docs/commercetools/api-composables-reference/modules.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# `API Reference - v1.1.0`
+
+## Table of contents
+
+### Modules
+
+- [getters/\_utils](modules/getters__utils.md)
```
Comment (filrak) [inline packages/core/docs/commercetools/api-composables-reference/modules.md:7]: there should be only composables here

**Review by filrak — CHANGES_REQUESTED** [review]

**Review by filrak — COMMENTED** [review]
feedback on a call

**Comment by lukaszjedrasik** [comment]
@filrak ready for review

File: packages/core/docs/commercetools/use-product.md [hunk 3]
```diff
@@ -2,19 +2,112 @@
...
   term?: any;
-  filters?: any;
+  filters?: Record<string, Filter>;
+  catId?: string | string[];
+  skus?: string[];
+  slug?: string;
+  id?: string;
+}
+```
+- `products` - a main data object that contains an array of products fetched by `search` method,
```
Comment (filrak) [inline packages/core/docs/commercetools/use-product.md:32]: missing type

File: packages/core/docs/commercetools/use-product.md [hunk 4]
```diff
@@ -2,19 +2,112 @@
...
+- `getFormattedPrice` - returns product price with currency sign.
+
+- `getTotalReviews` - returns total number of reviews product has. 
+
+- `getAverageRating` - returns average rating from all reviews.
+
+Interface for the above getter looks like this:
+
+```ts
+interface ProductGetters<PRODUCT, PRODUCT_FILTER> {
```
Comment (filrak) [inline packages/core/docs/commercetools/use-product.md:71]: use commercetools types

File: packages/core/docs/commercetools/use-category.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+## Getters
+
+Because `categories` property is a raw response, it's recommended to use `categoryGetters` for accessing any data from it. It includes following helper functions:
+
+- `getTree` - returns category tree.
+
+Interface for the above getter looks like this:
+
+```ts
+interface CategoryGetters<CATEGORY> {
```
Comment (filrak) [inline packages/core/docs/commercetools/use-category.md:34]: use commercetools types

**Review by filrak — CHANGES_REQUESTED** [review]

File: packages/core/docs/commercetools/use-review.md [hunk 6]
```diff
@@ -2,12 +2,187 @@
...
+  getReviewDate: (item: Review) => string;
+}
+
+interface AgnosticRateCount {
+  rate: number;
+  count: number;
+}
+```
+
+## Usage
```
Comment (filrak) [inline packages/core/docs/commercetools/use-review.md:97]: redundant

File: packages/core/docs/commercetools/use-cart.md [hunk 7]
```diff
@@ -2,9 +2,21 @@
 platform: Commercetools
```
Comment (filrak) [inline packages/core/docs/commercetools/use-cart.md:2]: we don't need this anymore

File: packages/core/docs/commercetools/use-cart.md [hunk 8]
```diff
@@ -2,9 +2,21 @@
...
 ---
 
-<IncludeContent content-key="use-cart" />
+# Cart
```
Comment (filrak) [inline packages/core/docs/commercetools/use-cart.md:5]:
```suggestion
# `useCart`
```

File: packages/core/docs/commercetools/use-cart.md [hunk 9]
```diff
@@ -47,10 +61,137 @@ type Cart = Versioned & {
...
+  
+- `applyCoupon` - function for applying coupon to cart
+  
+- `removeCoupon` - function for removing coupon applied to cart
+  
+- `loading` - a reactive object containing information about loading state of the cart
+
+## Getters
+
+Because `cart` property is a raw response with some additional properties, it's recommended to use `cartGetters` for accessing any data from it. It includes the following helper functions:
```
Comment (filrak) [inline packages/core/docs/commercetools/use-cart.md:90]:
```suggestion
```

File: packages/core/docs/commercetools/use-cart.md [hunk 10]
```diff
@@ -47,10 +61,137 @@ type Cart = Versioned & {
...
+
+- `getTotalItems` - returns products amount.
+
+- `getFormattedPrice` - returns product price with currency sign.
+
+- `getCoupons` - returns applied coupons.
+
+- `getDiscounts` - returns all discounts.
+
+Interface for the above getter looks like this:
```
Comment (filrak) [inline packages/core/docs/commercetools/use-cart.md:118]:
```suggestion
```

File: packages/core/docs/commercetools/use-cart.md [hunk 11]
```diff
@@ -47,10 +61,137 @@ type Cart = Versioned & {
...
+  description: string;
+  value: number;
+  code?: string;
+}
+```
+
+## Examples
+
+```js
+import { useCart, cartGetters } from '@vue-storefront/commercetools';
```
Comment (filrak) [inline packages/core/docs/commercetools/use-cart.md:174]: for enterprise ones change that to `@vsf-enterprise/commercetools`

File: packages/core/docs/commercetools/use-cart.md [hunk 12]
```diff
@@ -47,10 +61,137 @@ type Cart = Versioned & {
...
+      products: computed(() => cartGetters.getItems(cart.value)),
+      totals: computed(() => cartGetters.getTotals(cart.value)),
+      totalItems: computed(() => cartGetters.getTotalItems(cart.value))
+    }
+  }
+}
 ```
-:::
 
 :::tip
```
Comment (filrak) [inline packages/core/docs/commercetools/use-cart.md:196]: redundant

File: packages/core/docs/commercetools/use-category.md [hunk 13]
```diff
@@ -0,0 +1,104 @@
+# `useCategory`
+
+## Features
+
+`useCategory` composable is responsible for fetching a list of categories. A common usage scenario for this composable is navigation.
+
+## API
+
+- `search` - a main querying function that is used to query categories from eCommerce platform and populate the `categories` object with the result. Every time you invoke this function API request is made. This method accepts a single `params` object. The `params` has the following options:
```
Comment (filrak) [inline packages/core/docs/commercetools/use-category.md:9]: missing parameters

File: packages/core/docs/commercetools/use-facet.md [hunk 14]
```diff
@@ -1,58 +1,173 @@
...
+  attributes: ProductType;
+  attributeList: Array<Attribute>;
+}
+```
 
-Example of the category page browsing.
+## Example
 
 ```js
 import { useFacet, facetGetters } from '@vue-storefront/your-integration';
```
Comment (filrak) [inline packages/core/docs/commercetools/use-facet.md:173]:
```suggestion
import { useFacet, facetGetters } from '@vsf-enterprise/commercetools';
```

File: packages/core/docs/commercetools/use-review.md [hunk 15]
```diff
@@ -1,13 +1,156 @@
...
+  rate: number;
+  count: number;
+}
+```
+
+## Example
+
+```typescript
+import { onSSR } from '@vue-storefront/core';
+import { useReview, reviewGetters } from '@vsf-enterprise/ct-reviews';
```
Comment (filrak) [inline packages/core/docs/commercetools/use-review.md:131]:
```suggestion
import { useReview, reviewGetters } from '@vsf-enterprise/commercetoolss';
```

File: packages/core/docs/commercetools/use-user-shipping.md [hunk 16]
```diff
@@ -1,13 +1,336 @@
...
+type ShippingAddress = Address & {
+  isDefault?: boolean;
+}
+```
+
+## Example
+
+```typescript
+import { onSSR } from '@vue-storefront/core';
+import { useUserShipping, userShippingGetters } from '@vsf-enterprise/ct-shipping';
```
Comment (filrak) [inline packages/core/docs/commercetools/use-user-shipping.md:313]:
```suggestion
import { useUserShipping, userShippingGetters } from '@vsf-enterprise/commercetools';
```

File: packages/core/docs/commercetools/use-user-billing.md [hunk 17]
```diff
@@ -1,13 +1,326 @@
...
+type BillingAddress = Address & {
+  isDefault?: boolean;
+}
+```
+
+## Example
+
+```typescript
+import { onSSR } from '@vue-storefront/core';
+import { useUserBilling, userBillingGetters } from '@vsf-enterprise/ct-billing';
```
Comment (filrak) [inline packages/core/docs/commercetools/use-user-billing.md:303]:
```suggestion
import { useUserBilling, userBillingGetters } from '@vsf-enterprise/commercetools';
```

File: packages/core/docs/commercetools/use-wishlist.md [hunk 18]
```diff
@@ -1,13 +1,278 @@
...
+  productSlug?: Maybe<Scalars["String"]>;
+  variant?: Maybe<ProductVariant>;
+}
+```
+
+## Example
+
+```typescript
+import { onSSR } from '@vue-storefront/core';
+import { useWishlist, wishlistGetters } from '@vsf-enterprise/ct-wishlist';
```
Comment (filrak) [inline packages/core/docs/commercetools/use-wishlist.md:259]:
```suggestion
import { useWishlist, wishlistGetters } from '@vsf-enterprise/commercetools';
```

**Review by filrak — APPROVED** [review]

**Pull request merged**